src/data/sym/card.lean

```diff
@@ -0,0 +1,85 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import algebra.big_operators.basic
+import data.sym.sym2
+
+/-!
+# Stars and bars
+
+In this file, we prove the case `k = 2` of stars and bars.
+
+## Informal statement
+
+If we have `n` objects to put in `k` boxes, we can do so in exactly `(n + k - 1).choose n` ways.
+
+## Formal statement
+
+We can identify the `k` boxes with the elements of a fintype `α` of card `k`. Then placing `n`
+elements in those boxes corresponds to choosing how many of each element of `α` appear in a multiset
+of card `n`. `sym α n` being the subtype of `multiset α` of multisets of card `n`, our type stars
+and bars can be written as
+```lean
+lemma stars_and_bars {α : Type*} [fintype α] (n : ℕ) :
+  card (sym α n) = (card α + n - 1).choose (card α) := sorry
+```
+
+## TODO
+
+Prove the general case of stars and bars.
+-/
+
+open finset fintype
+
+namespace sym2
+variables {α : Type*} [decidable_eq α]
+
+/-- The `off_diag` of `s : finset α` is sent on a finset of `sym2 α` of card `s.card.choose 2`. -/
+lemma card_image_off_diag (s : finset α) :
+  (s.off_diag.image quotient.mk).card = s.card.choose 2 :=
```

Comment:
In my opinion this statement would be better as
```suggestion
  (s.off_diag.image quotient.mk).card = s.off_diag.card / 2 :=
```
which essentially just means moving the `rw` on line 43 to `card_sym2_not_diag`

Reply:
Or even
```suggestion
  2 * (s.off_diag.image quotient.mk).card = s.off_diag.card :=
```
since `nat.div` is painful and removing the `/2` in another lemma would require first proving `s.off_diag` is even.

Reply:
I think this version is useful, but the existing version should exist too, especially since the generalisation involves `choose`.

Reply:
Sure, a separate `card_image_off_diag_eq_choose` proved in terms of `card_image_off_diag` seems fine to me.

Reply:
But note that we end up rewriting away the choose again to prove the `card` lemma in this file, so I don't think `card_image_off_diag_eq_choose` is actually convenient to use in `card.

Reply:
I'm not so sure it's an improvement, because you might want the numerical result directly, but I've still done the change for now.

Reply:
Oh whoops I missed out on all the conversation above. I'm changing that again.